02-peer-protocol.md

```diff
@@ -50,10 +68,402 @@ pubkey corresponding to the funding output nothing prevents duplicative channel
 ids.
 
 
-## Channel Establishment
+### `channel_id`, v2
+
+For channels established using the v2 protocol, the `channel_id` is the
+SHA256(lesser-revocation-basepoint || greater-revocation-basepoint),
+where the lesser and greater is based off the order of the basepoint.
+
+If the peer's revocation basepoint is unknown (e.g. `open_channel2`
+and `accept_channel2`), a temporary `channel_id` should be found by
+using a zeroed out basepoint for the unknown peer.
+
+#### Rationale
+These values must be remembered by both peers for correct operation anyway.
+They're known from the first exchange of messages, obviating the need for
+a `temporary_channel_id`. Finally, by mixing information from both sides,
+they avoid `channel_id` collisions.
+
+
+## Interactive Transaction Construction
+
+Interactive transaction construction allows two peers to collaboratively
+build a transaction for broadcast.  This protocol is the foundation
+for dual-funded channels establishment (v2).
+
+### Set-Up and Vocabulary
+
+There are two parties to a transaction construction: an *initiator*
+and a *non-initiator*.
+ The *initiator* is the peer which initiates the protocol, e.g.
+for channel establishment v2 the *initiator* would be the peer which
+sends `open_channel2`; for a close, the *initiator* sends `shutdown`.
```

Comment:
I don't think the characterization works for a close, because both parties send `shutdown`, and they can do so concurrently.